src/agent/src/main.cpp

```diff
@@ -1,8 +1,53 @@
 #include "agent.hpp"
+#include "cmd_ln_parser.hpp"
+#include "register.hpp"
+
 #include <iostream>
+#include <optional>
 
-int main()
+int main(int argc, char* argv[])
 {
+    CommandlineParser cmdParser(argc, argv);
+
+    if (cmdParser.OptionExists("--register"))
+    {
+        std::cout << "Starting registration process" << std::endl;
+
+        if (cmdParser.OptionExists("--user") && cmdParser.OptionExists("--password"))
+        {
+            const auto user = cmdParser.getOptionValue("--user");
+            const auto password = cmdParser.getOptionValue("--password");
+            std::optional<std::string> name;
```

Comment:
We can make both `name` and `ip` variables `const` by using ternary operators:

```c++
const std::optional<std::string> name = cmdParser.OptionExists("--name") ? cmdParser.getOptionValue("--name") : std::nullopt;
```

Or lambdas:

```c++
const std::optional<std::string> name = [=] {
    if (cmdParser.OptionExists("--name"))
    {
        return cmdParser.getOptionValue("--name");
    }
    return std::nullopt;
}();
```

In this case the ternary operator may be more straight forward and enough for it.

Reply:
Done with ternary operator!